bootstrap/internal/controller/agentbootstrapconfig_controller.go

```diff
@@ -294,6 +312,28 @@ func (r *AgentBootstrapConfigReconciler) isReferencingACI(clusterDeployment *hiv
 	return clusterDeployment.Spec.ClusterInstallRef != nil
 }
 
+func (r *AgentBootstrapConfigReconciler) handleDeletion(ctx context.Context, config *bootstrapv1beta1.AgentBootstrapConfig, owner *bsutil.ConfigOwner) (ctrl.Result, error) {
```

Comment:
Theoretically, the metal3machine takes care of deprovisioning the BMH. Not sure what we should do about the agent?

Reply:
Let's say a user wants to scale down, what will happen?
Replicas are adjusted either in control plane or machine deployment. Machine are going to be deleted, which in turn will probably delete metal3machine, which will delete the BMH.

What would happen to agentbootstrapconfig? Is machine controller going to issue a delete to agentboostrapconfig? If so, should we safely remove the node (agent) from the cluster?

I think it's reasonable to think of deletion as scaling down

Reply:
As discussed offline we will focus on SNO: the machine deletion is going to cause BMH to get deprovisioned, so cleaning up the agent CR has no side effect.